core/vm/logger.go

```diff
@@ -200,6 +206,72 @@ func (l *StructLogger) CaptureState(pc uint64, op OpCode, gas, cost uint64, scop
 			)
 			l.storage[contract.Address()][address] = value
 			storage = l.storage[contract.Address()].Copy()
+
+			proof, err = l.env.StateDB.GetStorageProof(contract.Address(), address)
+			if err != nil {
+				log.Warn("Failed to get proof", "contract address", contract.Address().String(), "key", address.String(), "err", err)
+			}
```

Comment:
why only for `SSTORE` but no for `SLOAD`

Reply:
SLOAD won't change storage, do we need add that?

Reply:
> SLOAD won't change storage

it won't. but we still need proof for it (otherwise if we only have `sload` but no `sstore` in a block, how do you know the `storageProof`)

By my answer of "doesn't mattter" in https://www.notion.so/L2geth-34ecc78de12c45348112c511f42f9e31?d=3be1aa921aaf47d8a34a04d22222855a#de9082e2cd3e4896b34ff8d40154ad7b I mean it doesn't matter before or after. but we need at least one

Reply:
I see, `SLOAD` missed.

Reply:
changed.